src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -95,7 +95,10 @@ public static ByteStringContext.InternalScope GetLower(ByteStringContext byteStr
 
                 // PERF: Trick to avoid multiple compare instructions on hot loops. 
                 //       This is the same as (ch >= 65 && ch <= 90)
-                if (ch - 65 <= 90 - 65) 
+                const byte max = 90 - 65;
+                // We need to use byte here, to avoid automatic promotion to int32
+                var remaining = (byte)(ch - 65);
+                if (remaining <= max)
```

Comment:
I think we also need to handle https://github.com/ravendb/ravendb/blob/9e63a0ad28db7bede401782cb2a882ef7d249e89/src/Raven.Server/Documents/DocumentIdWorker.cs#L214 ?

Reply:
Inside `GetLowerIdSliceAndStorageKey`

Reply:
yes!